packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -1,292 +1,279 @@
+    }
+    if (extensionState.values.refundAddress) {
+      throw Error(`Refund address already given`);
+    }
+    if (!requestState.payer) {
+      throw Error(`The request must have a payer`);
+    }
+    if (!Utils.identity.areEqual(actionSigner, requestState.payer)) {
+      throw Error(`The signer must be the payer`);
+    }
 
-  // update payment address
-  copiedExtensionState.values.paymentAddress = extensionAction.parameters.paymentAddress;
-  // update events
-  copiedExtensionState.events.push({
-    name: ExtensionTypes.PnAddressBased.ACTION.ADD_PAYMENT_ADDRESS,
-    parameters: { paymentAddress: extensionAction.parameters.paymentAddress },
-    timestamp,
-  });
+    const copiedExtensionState: ExtensionTypes.IState = Utils.deepCopy(extensionState);
 
-  return copiedExtensionState;
-}
+    // update refund address
+    copiedExtensionState.values.refundAddress = extensionAction.parameters.refundAddress;
+    // update events
+    copiedExtensionState.events.push({
+      name: ExtensionTypes.PnAddressBased.ACTION.ADD_REFUND_ADDRESS,
+      parameters: { refundAddress: extensionAction.parameters.refundAddress },
+      timestamp,
+    });
 
-/**
- * Applies add refund address
- *
- * @param isValidAddress address validator function
- * @param extensionState previous state of the extension
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- *
- * @returns state of the extension updated
- */
-function applyAddRefundAddress(
-  isValidAddress: (address: string) => boolean,
-  extensionState: ExtensionTypes.IState,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): ExtensionTypes.IState {
-  if (
-    extensionAction.parameters.refundAddress &&
-    !isValidAddress(extensionAction.parameters.refundAddress)
-  ) {
-    throw Error('refundAddress is not a valid address');
-  }
-  if (extensionState.values.refundAddress) {
-    throw Error(`Refund address already given`);
-  }
-  if (!requestState.payer) {
-    throw Error(`The request must have a payer`);
-  }
-  if (!Utils.identity.areEqual(actionSigner, requestState.payer)) {
-    throw Error(`The signer must be the payer`);
+    return copiedExtensionState;
   }
 
-  const copiedExtensionState: ExtensionTypes.IState = Utils.deepCopy(extensionState);
-
-  // update refund address
-  copiedExtensionState.values.refundAddress = extensionAction.parameters.refundAddress;
-  // update events
-  copiedExtensionState.events.push({
-    name: ExtensionTypes.PnAddressBased.ACTION.ADD_REFUND_ADDRESS,
-    parameters: { refundAddress: extensionAction.parameters.refundAddress },
-    timestamp,
-  });
-
-  return copiedExtensionState;
+  /**
+   * Validate the extension action regarding the currency and network
+   * It must throw in case of error
+   *
+   * @param request
+   */
```

Comment:
```suggestion
```

Shouldn't we keep the documentation for when a child does something very specific compared to a parent?